### PR TITLE
PT-1287  Integrate Tidepool master for platform

### PR DESCRIPTION
**Pull request by CedricPerotto**
There is two new environment variable now needed for platform-user service (see .vscode/launch.json) :
- TIDEPOOL_USER_STORE_USERNAME
- TIDEPOOL_USER_STORE_PASSWORD

This means that the service until now couldn't connect to mongodb (and now it does on startup to ensure indexes are created)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,32 @@
+module github.com/tidepool-org/platform
+
```
Comment (yvon-dblg) [inline go.mod:2]: Maybe do the same like other go projects ?
```
replace github.com/tidepool-org/platform=> ./
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
 # Platform 
 The Tidepool platform API
 
+## Unreleased
+### Changed
```
Comment (yvon-dblg) [inline CHANGELOG.md:5]: Maybe the Tidepool changelog below is no longer relevant ?
Since they have stopped to update it

**Review by yvon-dblg — COMMENTED** [review]
The env.sh file seems to target styx URL.
Should we be worried ?

**Comment by yvon-dblg** [comment]
It's strange to have the deps in vendors, even it they seems to not be used.
Was the purpose of using `go get` and `go.mod` was to fetch them ?
Maybe for theses SOUPS we should use Nexus as well ?

File: go.mod (hunk 1 above)
Comment (CedricPerotto) [reply go.mod:2]: Because it is useless. Implicitly all packages path are relative to the path of the go.mod file.

File: CHANGELOG.md (hunk 2 above)
Comment (CedricPerotto) [reply CHANGELOG.md:5]: I don't understand this point...

**Comment by CedricPerotto** [comment]
> The env.sh file seems to target styx URL.
Should we be worried ?

Yes we should... At least platform-auth service joins shoreline through the gateway (and all metrics call did also but we stripped highwater from platform).

**Comment by CedricPerotto** [comment]
> It's strange to have the deps in vendors, even it they seems to not be used.
Was the purpose of using go get and go.mod was to fetch them ?
Maybe for theses SOUPS we should use Nexus as well ?

The current build (through the Makefile used in travis) doesn't use modules (no GO11MODULE is exported). 
Many instructions in the Makefile (lint, vet, imports...) are based on the vendor path so the vendor path is still used.
I think go.mod is used to dump the vendor tree (through `go mod vendor`  command) just like before when dep was used (through `go dep ensure` command). But I'm not aware about Tidepool dev process.

File: CHANGELOG.md (hunk 2 above)
Comment (yvon-dblg) [reply CHANGELOG.md:5]: There is a long list of changes made by tidepool, below our changelog.
But this changelist seems not being maintain anymore.

Comment (CedricPerotto) [reply CHANGELOG.md:5]: Yes but that's the point of a changelog keeping history of what's been done... So why remove it now ?

**Review by yvon-dblg — APPROVED** [review]
LGTM

**Pull request merged**